packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/datadog-agent.ts

```diff
@@ -0,0 +1,182 @@
+      environment.DD_TRACE_ANALYTICS_ENABLED = 'true';
+      environment.DD_SERVICE = this.props.serviceName;
+    }
+
+    return {
+      ...props,
+      environment,
+    } as ecs.ContainerDefinitionOptions;
+  }
+}
+
+/**
+ * Settings for the Datadog agent
+ **/
+export interface DatadogExtensionProps {
+  /**
+   * Whether or not to enable APM tracing
+   * @default false
+   */
+  readonly apmEnabled?: boolean,
+
+  /**
+   * Whether or not to enable trace analytics
+   * @default false
+   */
+  readonly traceAnalyticsEnabled?: boolean
+
+  /**
+   * An ecs.Secret that is storing the Datadog API key.
+   */
+  readonly datadogApiKey: ecs.Secret
+
+  /**
+   * Whether or not to capture the application logs
+   */
+  readonly logCaptureEnabled?: boolean
+
+  /**
+   * The version of the Datadog agent to use
+   * @default - DatadogAgentVersion.DATADOG_AGENT_7
+   */
+  readonly agentVersion?: DatadogAgentVersion
+}
+
+/**
+ * This extension adds a Datadog agent to the task definition and
+ * configures the primary container to be able to connect to it
+ */
+export class DatadogAgent extends ServiceExtension {
+  private apmEnabled: boolean;
+  private traceAnalyticsEnabled: boolean;
+  private datadogApiKey: ecs.Secret;
+  private agentVersion: DatadogAgentVersion;
+
+  constructor(props: DatadogExtensionProps) {
+    super('datadogAgent');
+
+    this.apmEnabled = props.apmEnabled === undefined ? false : props.apmEnabled;
+    this.traceAnalyticsEnabled = props.traceAnalyticsEnabled === undefined ? false : props.traceAnalyticsEnabled;
+    this.agentVersion = props.agentVersion === undefined ? DatadogAgentVersion.DATADOG_AGENT_7 : props.agentVersion;
+    this.datadogApiKey = props.datadogApiKey;
+  }
+
+  public prehook(service: Service, scope: cdk.Construct) {
+    this.parentService = service;
+    this.scope = scope;
+  }
+
+  // If some of the Datadog settings require setting env variables on the
+  // Datadog client side SDK, then attach a mutating hook to the primary
+  // container to do that.
+  public addHooks() {
+    if (!this.traceAnalyticsEnabled) {
+      return;
+    }
+
+    const container = this.parentService.serviceDescription.get('service-container') as Container;
+
+    container.addContainerMutatingHook(new DatadogAgentMutatingHook({
+      traceAnalyticsEnabled: this.traceAnalyticsEnabled,
+      serviceName: this.parentService.id,
+    }));
+  }
+
+  // This hook adds and configures the Datadog agent in the task.
+  public useTaskDefinition(taskDefinition: ecs.TaskDefinition) {
+    let envVariables: any = {
+      // Tag any data captured by with the name of the environment
+      DD_ENV: this.parentService.environment.id,
+
+      // This is running in an ECS/Fargate env
+      ECS_FARGATE: 'true',
+    };
+
+    if (this.apmEnabled) {
+      envVariables.DD_APM_ENABLED = 'true';
+      envVariables.DD_APM_NON_LOCAL_TRAFFIC = 'true';
+    }
+
+    envVariables.DD_ECS_COLLECT_RESOURCE_TAGS_EC2 = 'true';
```

Comment:
I tested this and it does not appear to have an adverse effects on Fargate. In fact it appears that it might just be legacy named, and causes Datadog to collect resource tags from Fargate as well as EC2 now